### PR TITLE
Add single functional test for creating a private group

### PR DESCRIPTION
**Pull request by lyzadanger**
We need more functional tests for our API endpoints. This PR represents the first step in this; it adds a single, simple functional test (a success-path test) for the recently-added create-group endpoint. It also moves groups-API tests under a subdirectory, `api`, as a starting point for a light reorg of these test modules.

It is my hope that having increased functional tests for our API endpoints will make our services more reliable and let us rest a little easier.

This is relevant to LMS because we are adding a number of API endpoints to support the project and extending how `auth_client` authentication works.

Basically:

* Does this look generally OK?
* Am I okay to add a bunch of functional tests? I know there has been some notion of "they run slow" in the past, but the entire suite runs in 11 seconds for me locally; I'll see how long they take to run on CI

We currently have zero tests for any `auth_client`-auth'd endpoint; that's probably my next step here.

### CODE REVIEW

**Comment by lyzadanger** [comment]
Update: current suite of functional tests took 19 seconds to run on Jenkins during the build for this branch.

File: tests/functional/api/test_groups.py [hunk 1]
```diff
@@ -8,6 +8,21 @@
 @pytest.mark.functional
 class TestGroupAPI(object):
 
+    def test_create_group(self, app, user_with_token):
+        """Test a request to create a private group through the API"""
+
+        user, token = user_with_token
```
Comment (seanh) [inline tests/functional/api/test_groups.py]: This is fine, but an alternative would be for `token` and `user` to be two separate fixtures, with the `token` fixture using the `user` fixure, then tests could just take `user, token` as separate test method parameters.

File: tests/functional/api/test_groups.py [hunk 2]
```diff
@@ -8,6 +8,21 @@
...
+        headers = {'Authorization': str('Bearer {}'.format(token.value))}
```
Comment (seanh) [inline tests/functional/api/test_groups.py]: I'm not sure what the `str` here is for?

File: tests/functional/api/test_groups.py [hunk 3]
```diff
@@ -44,3 +59,12 @@ def group_member_with_token(group_member, db_session, factories):
...
     db_session.commit()
     return (group_member, token)
+
+
+@pytest.fixture
+def user_with_token(db_session, factories):
+    user = factories.User()
+    token = factories.DeveloperToken(userid=user.userid)
+    db_session.add(token)
+    db_session.commit()
```
Comment (seanh) [inline tests/functional/api/test_groups.py]: I _think_ the `add()` here is unnecessary (`user` will have been added automatically by the factory, and `token` will have been added because it's attached to `user`).
